akka-stream-tests/src/test/scala/akka/stream/io/TlsSpec.scala

```diff
@@ -148,29 +163,38 @@ class TlsSpec extends StreamSpec("akka.loglevel=INFO\nakka.actor.debug.receive=o
     object ClientInitiatesViaTcp extends CommunicationSetup {
       var binding: Tcp.ServerBinding = null
       def decorateFlow(leftClosing: TLSClosing, rightClosing: TLSClosing,
-                       rhs: Flow[SslTlsInbound, SslTlsOutbound, Any]) = {
-        binding = server(serverTls(rightClosing).reversed join rhs)
+                       rhs: Flow[SslTlsInbound, SslTlsOutbound, Any], ks: SharedKillSwitch) = {
+        binding = server(serverTls(rightClosing).reversed join rhs.via(ks.flow))
         clientTls(leftClosing) join Tcp().outgoingConnection(binding.localAddress)
       }
-      override def cleanup(): Unit = binding.unbind()
+      override def cleanup(): Unit = {
+        super.cleanup()
+        binding.unbind()
+      }
     }
 
     object ServerInitiatesViaTcp extends CommunicationSetup {
       var binding: Tcp.ServerBinding = null
       def decorateFlow(leftClosing: TLSClosing, rightClosing: TLSClosing,
-                       rhs: Flow[SslTlsInbound, SslTlsOutbound, Any]) = {
-        binding = server(clientTls(rightClosing).reversed join rhs)
+                       rhs: Flow[SslTlsInbound, SslTlsOutbound, Any], ks: SharedKillSwitch) = {
+        binding = server(clientTls(rightClosing).reversed join rhs.via(ks.flow))
         serverTls(leftClosing) join Tcp().outgoingConnection(binding.localAddress)
       }
-      override def cleanup(): Unit = binding.unbind()
+      override def cleanup(): Unit = {
+        super.cleanup()
```

Comment:
Thank you for pointing out. I forgot to remove this after one of the test implementation variants